hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/engine/StepRunner.java

```diff
@@ -217,8 +216,9 @@ private void offloadRun(Runnable step,
             operationExecutor.execute(offload);
         } else {
             Offloadable offloadableStep = (Offloadable) step;
-            executorName = offloadableStep.getExecutorName();
-            executionService.getExecutor(executorName).execute(offload);
+            currentExecutorName = offloadableStep.getExecutorName();
+            // TODO what if there is RejectedExecutionException
```

Comment:
Let's either resolve the TODO or create an issue.

Reply:
it is here https://github.com/hazelcast/hazelcast/pull/22684